### PR TITLE
Change how projects are opened and tracked

### PR DESCRIPTION
**Pull request by cgay**
...and other changes.

### CODE REVIEW

File: compiler.dylan [hunk 1]
```diff
@@ -26,11 +22,8 @@ define function run-compiler(server, string :: <string>) => ()
 end function run-compiler;
 
 // Ask the command line compiler to open a project.
-// Param: server - the command line server.
-// Param: name - either a library name or a lid file.
-// Returns: an instance of <project-object>
```
Comment (pedro-w) [inline compiler.dylan:31]: Is there a policy for whether these kind of comments should be included or not?

File: lsp-dylan.dylan [hunk 2]
```diff
@@ -354,6 +324,10 @@ define function publish-diagnostics
                            "diagnostics", diagnostics));
   end;
   // Clear diagnostics for URIs that no longer have any.
+  //
+  // TODO(cgay): this isn't working, in emacs at least.  An alternative would
+  // be to clear diagnostics for all $documents that aren't in warnings-by-uri,
```
Comment (pedro-w) [inline lsp-dylan.dylan]: Based on quite limited testing it does work on VScode and emacs too (appeared and disappeared from the *Flymake diagnostics for '...'* buffer)

**Review by pedro-w — APPROVED** [review]

File: compiler.dylan (hunk 1 above)
Comment (cgay) [reply compiler.dylan:31]: I just thought putting the types in the signature obviated the need for the comments. (But now I notice I didn't add the `<project-object>` return type as I meant to, which I'll fix.) Please take another look.

File: lsp-dylan.dylan (hunk 2 above)
Comment (cgay) [reply lsp-dylan.dylan]: ok, I removed the TODO.  If it really doesn't work under some circumstances I'm sure we'll notice anyway. I tend to over-use TODOs for whatever problem I have on the brain at the time.

**Pull request closed without merging**